src/app/clusters/ota-requestor/ota-requestor-server.cpp

```diff
@@ -78,6 +81,67 @@ CHIP_ERROR OtaSoftwareUpdateRequestorAttrAccess::Write(const ConcreteDataAttribu
 
 } // namespace
 
+namespace chip {
+
+// -----------------------------------------------------------------------------
+// OtaRequestorServer implementation
+
+static OtaRequestorServer sInstance;
+
+OtaRequestorServer & OtaRequestorServer::GetInstance(void)
+{
+    return sInstance;
+}
+
+EmberAfStatus OtaRequestorServer::SetUpdateState(OTAUpdateStateEnum value)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    // Find all endpoints that has OtaSoftwareUpdateRequestor implemented
+    for (auto endpoint : EnabledEndpointsWithServerCluster(OtaSoftwareUpdateRequestor::Id))
+    {
+        OTAUpdateStateEnum currentValue;
+        status = Attributes::UpdateState::Get(endpoint, &currentValue);
+        VerifyOrDie(EMBER_ZCL_STATUS_SUCCESS == status);
+
+        if (currentValue != value)
+        {
+            status = Attributes::UpdateState::Set(endpoint, value);
+            VerifyOrDie(EMBER_ZCL_STATUS_SUCCESS == status);
+        }
+    }
+
+    return status;
+}
+
+EmberAfStatus OtaRequestorServer::SetUpdateStateProgress(uint8_t value)
+{
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+
+    // Find all endpoints that has OtaSoftwareUpdateRequestor implemented
+    for (auto endpoint : EnabledEndpointsWithServerCluster(OtaSoftwareUpdateRequestor::Id))
+    {
+        app::DataModel::Nullable<uint8_t> currentValue;
+        status = Attributes::UpdateStateProgress::Get(endpoint, currentValue);
+        VerifyOrDie(EMBER_ZCL_STATUS_SUCCESS == status);
+        if (!currentValue.IsNull())
+        {
+            if (currentValue.Value() != value)
```

Comment:
Shouldn't this be more like:
```suggestion
        if (currentValue.IsNull() || currentValue.Value() != value) {
```
?  Otherwise if currentValue is null we will never update the state, seems like.

Reply:
Yes, thanks good catch.